src/main/java/com/singh/astha/medicinereminder/services/impl/DosageServiceImpl.java

```diff
@@ -33,19 +38,22 @@ public class DosageServiceImpl implements DosageService {
     private final DosageDtoTransformer dosageDtoTransformer;
     private final DosageHistoryRepository dosageHistoryRepository;
     private final MedicineDtoTransformer medicineDtoTransformer;
+    private final EventReminderRepository eventReminderRepository;
 
     public DosageServiceImpl(MedicineRepository medicineRepository,
                              DosageDtoTransformer dosageDtoTransformer,
                              DosageHistoryRepository dosageHistoryRepository,
-                             MedicineDtoTransformer medicineDtoTransformer) {
+                             MedicineDtoTransformer medicineDtoTransformer,
+                             EventReminderRepository eventReminderRepository) {
         this.medicineRepository = medicineRepository;
         this.dosageDtoTransformer = dosageDtoTransformer;
         this.dosageHistoryRepository = dosageHistoryRepository;
         this.medicineDtoTransformer = medicineDtoTransformer;
+        this.eventReminderRepository = eventReminderRepository;
     }
 
     @Override
-    public MedicineResponseDto updateDosage(Long userId, DosageHistoryRequestDto dosageHistoryRequestDto) {
+    public MedicineResponseDto updateDosage(Long userId, DosageHistoryRequestDto dosageHistoryRequestDto) throws ParseException {
         DosageHistory dosageHistory;
         try {
             dosageHistory = dosageDtoTransformer.convertDosageHistoryRequestDtoToDosageHistory(
```

Comment:
Don't catch the `Exception` here. Instead catch IllegalArgument Exception for this usecase.